core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -201,18 +198,19 @@ public synchronized void connect() throws AlluxioStatusException {
             getServiceName(), retryPolicy.getAttemptCount(), e.toString());
         continue;
       }
-      // Bootstrap once for clients
-      if (!isConnected()) {
-        try {
-          Configuration.loadClusterDefault(mAddress);
-        } catch (UnavailableException e) {
-          LOG.warn("Failed to handshake ({}) with {} @ {}: {}", retryPolicy.getAttemptCount(),
-              getServiceName(), mAddress, e.getMessage());
-          continue;
-        }
-      }
       try {
-        doConnect();
+        beforeConnect();
```

Comment:
is moving this inside the try intentional? before it was outside

Reply:
intentional. So the beforeConnect and afterConnect are treated symmetrically.